core/src/main/java/hudson/Util.java

```diff
@@ -901,6 +916,23 @@ public static String encode(@NonNull String s) {
      */
     @NonNull
     public static String rawEncode(@NonNull String s) {
+        return encode(s, uriMap);
+    }
+
+    /**
+     * Encode a single path component for use in an HTTP URL.
+     * Escapes all special characters including those outside
+     * of the characters specified in RFC1738.
+     * Encoded characters are space an {@code "!#%$'&*()+,-./=:;< >?@_[\]^ `{|}~}
+     * Note that slash ({@code /}) is encoded, so the given string should be a
+     * single path component used in constructing a URL.
+     */
```

Comment:
```suggestion
     * @since TODO
     */
```

(Unless we prefer `@Restricted(NoExternalUse.class)`)